File: Adv_DataStructures/Reverse_str.py
def backwards_string(s : str) -> str:
    if not s:
        return ""
    return s[-1] + backwards_string(s[:len(s)-1])

if __name__ == '__main__':
    userInput = input("Enter a string: ")
    print(f"Reversed string is '{backwards_string(userInput)}'")

File: Adv_DataStructures/reverse_str.cpp
#include <iostream>

std::string backwards_string(std::string s){
    if(s == ""){
        return "";
    }
    return s.back() + backwards_string(s.substr(0, s.size()-1));
}

int main(){
    std::string userInput;
    std::cout << "Enter a string: ";
    std::cin >> userInput;

    std::cout << "Reversed string is " << "'"<< backwards_string(userInput) << "'" << std::endl;
    return 0;
}